backends/csvq.go

```diff
@@ -0,0 +1,230 @@
+package backends
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+)
+
+// csvqDB represents the csvqDB backend.
+type csvqDB struct {
+	SqlDB
+}
+
+// csvqDBWriter represents a writer that saves results
+// to a csvqDB backend.
+type csvqDBWriter struct {
+	sqlDBWriter
+}
```

Comment:
Here it's the tricky part ... this new backend implements the actual... but overwrite the only needed functionality...